lti_consumer/lti13.py

```diff
@@ -0,0 +1,187 @@
+
+    # LTI Claim version
+    # http://www.imsglobal.org/spec/lti/v1p3/#lti-version-claim
+    "https://purl.imsglobal.org/spec/lti/claim/version": "1.3.0",
+
+    # Optional claims - useless ones
+    # "https://purl.imsglobal.org/spec/lti/claim/context": {
+    #     "id": "c1d887f0-a1a3-4bca-ae25-c375edcc131a",
+    #     "label": "ECON 1010",
+    #     "title": "Economics as a Social Science",
+    #     "type": ["http://purl.imsglobal.org/vocab/lis/v2/course#CourseOffering"]
+    # },
+    # "https://purl.imsglobal.org/spec/lti/claim/tool_platform": {
+    #     "guid": "ex/48bbb541-ce55-456e-8b7d-ebc59a38d435",
+    #     "contact_email": "support@platform.example.edu",
+    #     "description": "An Example Tool Platform",
+    #     "name": "Example Tool Platform",
+    #     "url": "https://platform.example.edu",
+    #     "product_family_code": "ExamplePlatformVendor-Product",
+    #     "version": "1.0"
+    # },
+
+    # Optional claims - Useful
+    # This is useful for error redirects
+    # http://www.imsglobal.org/spec/lti/v1p3/#launch-presentation-claim
+    "https://purl.imsglobal.org/spec/lti/claim/launch_presentation": {
+        "document_target": "iframe",  # iframe, frame, window
+        # Endpoint to redirect the user to after completing LTI task
+        # Returns with log statements
+        "return_url": "https://platform.example.edu/terms/201601/courses/7/sections/1/resources/2"
+    },
+    # Custom variables :)
+    "https://purl.imsglobal.org/spec/lti/claim/custom": {
+        "xstart": "2017-04-21T01:00:00Z",
+        "request_url": "https://tool.com/link/123"
+    }
+}
+
+
+class LtiConsumer1p3:
+    def __init__(
+            self,
+            iss,
+            lti_oidc_url,
+            lti_launch_url,
+            deployment_id,
+            rsa_key,
+    ):
+        self.iss = iss
+        self.oidc_url = lti_oidc_url
+        self.launch_url = lti_launch_url
+        self.deployment_id = deployment_id
+
+        # Generate JWK from RSA key
+        self.jwk = RSAKey(
+            # Don't hardcode key name
+            kid="lti_key",
+            key=RSA.import_key(rsa_key)
+        )
+
+    def _encode_and_sign(self, message):
+        # Dump JSON and encode it with key
+        msg = json.dumps(message)
+
+        # The class instance that sets up the signing operation
+        # An RS 256 key is required for LTI 1.3
+        _jws = JWS(msg, alg="RS256")
+
+        # Encode and sign LTI message
+        return _jws.sign_compact([self.jwk])
+
+    @staticmethod
+    def _get_user_roles(roles):
+        """
+        Converts platform roles into LTI compliant roles
+
+        Used in roles claim: should return array of URI values
+        for roles that the user has within the message's context.
+
+        Reference: http://www.imsglobal.org/spec/lti/v1p3/#roles-claim
+        Role vocabularies: http://www.imsglobal.org/spec/lti/v1p3/#role-vocabularies
+        """
+        return [
```

Comment:
WIP

Reply:
Some seems pretty much a 1:1 reference to roles in edX.

Reply:
Not wip anymore, now there's a role mapping happening.